src/flowMC/strategy/adapt_step_size.py

```
@@ -14,6 +15,15 @@
 logger = logging.getLogger(__name__)
 
 
+@runtime_checkable
+class _PerDimensionAdaptableProposal(Protocol):
+    """Structural interface required by ``AdaptStepSizePerDimension``."""
+
+    def get_effective_dim_profile(self) -> Float[Array, " n_dim"]: ...
```

Ruff (0.16.0)

[error] Syntax error in forward annotation: Expected an expression

(F722)